docs/the-new-architecture/pillars-fabric-components.md

```diff
@@ -3,17 +3,37 @@ id: pillars-fabric-components
 title: Fabric Components
 ---
 
-This section contains a high-level introduction to Fabric components. It provides enough context to understand when a Fabric component is needed and how it roughly works.
-It points to the [Renderer](https://reactnative.dev/architecture/fabric-renderer) section of the [Architecture](https://reactnative.dev/architecture/overview) tab for a deep dive into the technical details.
+A Fabric Component is a UI component which is rendered on the screen using the [Fabric Renderer](https://reactnative.dev/architecture/fabric-renderer).
 
-This section must have a warning that it works only with the new architecture enabled. It points to the [migration section](../new-architecture-intro).
+Using Fabric Components instead of Native Components allows us to reap all the [benefit](./why) coming from the **New Architecture**. Specifically, we are able to leverage JSI to efficiently connect the Native UI code and the JavaScript one, and all the other features provided by the new renderer.
+
+A Fabric Component is created starting from a **JS specification**. This, with the help of a bit of [**Codegen**](./pillars-codegen), will create some C++ code, shared among all the RN's platforms. Then, after filling in the missing **native code** with Component specific logic, the Component can be **integrated** in the app.
```

Comment:
```suggestion
A Fabric Component is created starting from a **JS specification**. This, with the help of [**Codegen**](./pillars-codegen), will create some C++ code, shared among all React Native platforms. Then, after filling in the missing **native code** with component-specific logic, the Component can be integrated in the app.
```

Explanations:
* "Codegen" is a specific tool, so "a bit of" doesn't quite work to my ear, we can just say "with the help of Codegen"
* "C++ code..." I might consider rephrasing this further to specifically say that the C++ is integrated with the platform-native code, it might be unclear to a new reader
* "component-specific logic" needs to be hyphenated, because "component-specific" is acting as [a compound adjective on the noun "logic"](https://www.grammarly.com/blog/hyphen/)
* "integrated" is not used as a new or special term here, I wouldn't bold it ("native code" is similar, but closer to a "special term" the way RN docs treat it)
* "filling in the missing native code" is an ambiguous phrase to me. What is missing, from where? If we're talking about just writing the native code to bridge between C++ and JS, I would phrase it as "adding code that bridges..." or "connecting", instead of "filling in missing", because "missing" sounds like something was supposed to be there but something went wrong.

Reply:
I tried to rephrase this bit. Let me know what do you think!